packages/@react-spectrum/s2/src/ComboBox.tsx

```diff
@@ -643,20 +643,24 @@ const ComboboxInner = forwardRef(function ComboboxInner(props: ComboBoxProps<any
           description={descriptionMessage}>
           {errorMessage}
         </HelpText>
-        <PopoverBase
+        <Popover
           hideArrow
           triggerRef={triggerRef}
           offset={menuOffset}
           placement={`${direction} ${align}` as Placement}
           shouldFlip={shouldFlip}
           UNSAFE_style={{
-            width: menuWidth ? `${menuWidth}px` : undefined,
-            // manually subtract border as we can't set Popover to border-box, it causes the contents to spill out
-            '--trigger-width': `calc(${triggerWidth} - 2px)`
+            // Subtract by 2 since these widths are set on the inner div rather than on the outermost div element that has
+            // a border
+            width: menuWidth ? `calc(${menuWidth}px - 2 * var(--s2-container-border-width))` : undefined,
+            '--trigger-width': `calc(${triggerWidth} - 2 * var(--s2-container-border-width))`
```

Comment:
unfortunate consequence of switching to Popover from PopoverBase

Reply:
not sure i understand this comment

wish we didn't need to use UNSAFE style to declare variables that are arbitrary and the result of calculations

wonder if we should consider a new style prop just for variables, that should be safe

Reply:
its because previously these width values would go to the outer div that had the border applied to it, but now that we use Popover these values go to the inner div since we need to modify that div's padding/overflow. I'd really like if there was just a single container div per say, but applying `overflow: auto` to the outer div will hide the popover arrow...